### ISSUE TITLE
Data Science Workspace

### ISSUE THREAD

**Issue by 4tikhonov** [IQSS/dataverse#2286]
We're developing different applications to analyze our data and I think it can be quite useful to create something like Data Science Workspace to share all tools for the community. 

We're mining a lot of interesting stuff like historical names of countries, different variables, etc directly from datasets stored in Dataverses and creating Open Data APIs based on the extracted data.

There is one of our examples:
1. dataset uploaded in Dataverse
http://dv.sandbox.socialhistoryservices.org/dataset.xhtml?persistentId=hdl:10622/XUH7BG
2. API delivering all historical names automatically mined from this dataset
http://clearance.sandbox.socialhistoryservices.org/collabs/boundaries
3.  iPython notebook with most of the processing steps to show the approach:
https://github.com/4tikhonov/geo/blob/master/Clio%20Infra%20Historical%20boundaries.ipynb

Basically I think it should be something like iPython notebook for Dataverse stuff. I'm really curios to see what people can do with our tools/notebooks on their own data.

**Comment by pdurbin** [comment]
@4tikhonov it's neat that you're creating new APIs. To be clear, I assume these APIs are something you're building on the side in Python or something... they make SQL calls into the database.

For IPython stuff see #2210.

**Comment by 4tikhonov** [comment]
Well, the point is that doesn't matter how it works on backend side (it can be file, SQL or noSQL), we just need API endpoint to be able to query it and combine results with other endpoints from different datasets. This is a way to slice a data in specific sets and create 'super' dataset from different data files uploaded in Dataverse. We also need data quality check for every dataset and it can be universal tool for every dataset.

**Comment by pdurbin** [comment]
@4tikhonov any recent thinking on this issue? I'd be happy to review a pull request if you have specific changes in mind. These days there's also https://github.com/IQSS/miniverse to make pull requests against if that's a better place. If it's ok to close this issue, that's fine too. 😄

**Comment by 4tikhonov** [comment]
Hi Phil, actually we want to contribute to miniverse as we have started small project to connect the dashboard to all other systems at DANS. @raprasad, for your attention also.

**Comment by pdurbin** [comment]
@4tikhonov great! I'd love to hear more about your project! Lemme know when there's a readme or whatever. 😄

**Comment by pdurbin** [comment]
@4tikhonov any news?

**Comment by djbrooke** [comment]
Hey @4tikhonov - I'm going to close this out. We're moving on Dataset External Tools (#5028) and we'd want any data science tools for datasets to follow that framework.
